src/shared/dissect-image.c

```diff
@@ -228,6 +229,31 @@ static int make_partition_devname(
 }
 #endif
 
+#if HAVE_BLKID || HAVE_LIBCRYPTSETUP
+static int dissected_partition_open(DissectedPartition *p) {
+        _cleanup_close_ int fd = -1;
+        const char *node;
+
+        assert(p);
+
+        if (!p->decrypted_node && p->mount_node_fd >= 0)
+                return 0; /* Already opened. */
+
+        node = ASSERT_PTR(p->decrypted_node ?: p->node);
+
+        fd = open(node, O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
+        if (fd < 0)
+                return -errno;
+
+        log_debug("Opened %s (fd=%i).", node, fd);
+
+        safe_close(p->mount_node_fd);
+        p->mount_node_fd = TAKE_FD(fd);
+
+        return 0;
```

Comment:
i think it would make sense to to compare the BLKGETDISKSEQ value of these partition devices with the one of the "whole" device here, to ensure we don't end up with wrong devices.

And beforehand we should check that what we just opened is actually a block device. 

And this call should probably not even try to open() if the "whole" node is not actually a block device itself.

Reply:
OK.
(But, you said in my other PRs, `dissect_image()` should work for non device fd...)